Content.Client/Gameplay/GameplayState.cs

```diff
@@ -26,73 +36,113 @@ public sealed class GameplayState : GameplayStateBase, IMainViewportState
 
             base.Shutdown();
-            Viewport.Dispose();
             // Clear viewport to some fallback, whatever.
             _eyeManager.MainViewport = UserInterfaceManager.MainViewport;
             _fpsCounter.Dispose();
             _uiManager.ClearWindows();
+            UnloadMainScreen();
         }
 
-        public override void FrameUpdate(FrameEventArgs e)
+        public void ReloadMainScreen()
         {
-            base.FrameUpdate(e);
+            if (_uiManager.ActiveScreen == null)
+            {
+                return;
+            }
 
-            Viewport.Viewport.Eye = _eyeManager.CurrentEye;
+            UnloadMainScreen();
+            LoadMainScreen();
+        }
 
-            // verify that the current eye is not "null". Fuck IEyeManager.
+        private void UnloadMainScreen()
+        {
+            _chatController.SetMainChat(false);
+            _menuController.UnloadButtons();
+            _uiManager.UnloadScreen();
+        }
 
-            var ent = _playerMan.LocalPlayer?.ControlledEntity;
-            if (_eyeManager.CurrentEye.Position != default || ent == null)
-                return;
+        private void LoadMainScreen()
+        {
+            var screenTypeString = _configurationManager.GetCVar(CCVars.UILayout);
+            if (!Enum.TryParse(screenTypeString, out ScreenType screenType))
+            {
+                screenType = default;
+            }
 
-            _entMan.TryGetComponent(ent, out EyeComponent? eye);
-            
-            if (eye?.Eye == _eyeManager.CurrentEye
-                && _entMan.GetComponent<TransformComponent>(ent.Value).WorldPosition == default)
-                return; // nothing to worry about, the player is just in null space... actually that is probably a problem?
+            switch (screenType)
+            {
+                case ScreenType.Default:
+                    _uiManager.LoadScreen<DefaultGameScreen>();
+                    break;
+                case ScreenType.Separated:
+                    _uiManager.LoadScreen<SeparatedChatGameScreen>();
+                    break;
+            }
 
-            // Currently, this shouldn't happen. This likely happened because the main eye was set to null. When this
-            // does happen it can create hard to troubleshoot bugs, so lets print some helpful warnings:
-            Logger.Warning($"Main viewport's eye is in nullspace (main eye is null?). Attached entity: {_entMan.ToPrettyString(ent.Value)}. Entity has eye comp: {eye != null}");
+            _chatController.SetMainChat(true);
+            _viewportController.ReloadViewport();
+            _menuController.LoadButtons();
+
+            // TODO: This could just be like, the equivalent of an event or something
+            _ghostController.UpdateGui();
+            _actionController.RegisterActionContainer();
+            _alertsController.SyncAlerts();
+            _hotbarController.ReloadHotbar();
```

Comment:
can this not just be an action/c# event that gets invoked and which all of these subscribe to, seems to be a pattern elsewhere in UI

Reply:
state startup happens before the call that occurs when a state is entered, so i can't use that unfortunately - and since the state entered call also gives controllers a reference to the current state, it ends up being that i have to manually do this

Reply:
well thats annoying

Reply:
i can try to do one thing though, and that's just move all this to its own controller

Reply:
actually could you do this in a separate UI controller made for this purpose on the state enter

Reply:
ok github didnt load that comment glad we had the same idea